Xamarin.Forms.Core/Easing.cs

```diff
@@ -94,5 +164,224 @@ public double Ease(double v)
+			}
+		}
+
+		private static class ElasticEquations
+		{
+			const double P = 0.3;
+			const double S = 0.075;
+
+			public static double EaseOut(double x)
+			{
+				if (DoubleExtensions.ApproximatelyEquals(x, 1)) return 1;
+
+				return Math.Pow(2, -10 * x) * Math.Sin((x - P / 4) * (2 * Math.PI) / P) + 1;
+			}
+
+			public static double EaseIn(double x)
+			{
+				if (DoubleExtensions.ApproximatelyEquals(x, 0)) return 0;
+				if (DoubleExtensions.ApproximatelyEquals(x, 1)) return 1;
+
+				return -(Math.Pow(2, 10 * (x -= 1)) * Math.Sin((x - S) * (2 * Math.PI) / P));
+			}
+		}
+
+		private static class ExponentialEquations
+		{
+			public static double EaseOut(double x)
+			{
+				return DoubleExtensions.ApproximatelyEquals(x, 1) ? 1 : -Math.Pow(2, -10 * x) + 1;
+			}
+
+			public static double EaseIn(double x)
+			{
+				return DoubleExtensions.ApproximatelyEquals(x, 0) ? 0 : Math.Pow(2, 10 * (x - 1));
+			}
+
+			public static double EaseInOut(double x)
+			{
+				if (DoubleExtensions.ApproximatelyEquals(x, 0)) return 0;
+				if (DoubleExtensions.ApproximatelyEquals(x, 1)) return 1;
+
+				if ((x / 2) < 1)
+					return 0.5 * Math.Pow(2, 10 * (x - 1));
+
+				return 0.5 * -Math.Pow(2, -10 * --x) + 2;
+			}
+		}
+
+		private static class QuadEquations
+		{
+			public static double EaseOut(double t)
+			{
+				return -1 * t * (t - 2);
+			}
+
+			public static double EaseIn(double t)
+			{
+				return t * t;
+			}
+
+			public static double EaseInOut(double x)
+			{
+				if ((x / 2) < 1)
+					return 0.5 * x * x;
+
+				return -0.5 * ((--x) * (x - 2) - 1);
+			}
+		}
+
+		private static class QuarticEquations
+		{
+			public static double EaseOut(double x)
+			{
+				return -1 * ((x - 1) * Math.Pow(x, 3) - 1);
+			}
+
+			public static double EaseIn(double x)
+			{
+				return Math.Pow(x, 4);
+			}
+
+			public static double EaseInOut(double x)
+			{
+				if ((x / 2) < 1)
+					return -0.5 * Math.Pow(x, 4);
+
+				return -0.5 * ((x -= 2) * Math.Pow(x, 3) - 2);
+			}
+		}
+
+		private static class QuinticEquations
+		{
+			public static double EaseOut(double x)
+			{
+				return ((x - 1) * Math.Pow(x, 4) + 1);
+			}
+
+			public static double EaseIn(double x)
+			{
+				return Math.Pow(x, 5);
+			}
+
+			public static double EaseInOut(double x)
+			{
+				if ((x / 2) < 1)
+					return 0.5 * Math.Pow(x, 5);
+
+				return 0.5 * ((x -= 2) * Math.Pow(x, 4) + 2);
+			}
+		}
+
+		private static class SinusoidalEquations
+		{
+			public static double Dampened(double x)
+			{
+				// y = a * sin (bx - c) + d
+				// a = amplitude
+				// b = period (stretch or compress)
+				// c = horizontal translation (move graph left or right)
+				// d = vertical translation
+				return Math.Sin((10 * x) - x / 2) * Math.Exp(-x);
+			}
+		}
+
+		private static class DoubleExtensions
+		{
+			public static bool ApproximatelyEquals(double x, double y)
+			{
+				var epsilon = Math.Max(Math.Abs(x), Math.Abs(y)) * 1E-15;
+				return Math.Abs(x - y) <= epsilon;
+			}
```

Comment:
Since you're only ever using `ApproximatelyEquals` to check for equality with one or zero, I'd suggest creating specific methods for both. Something like:

```
public static class DoubleExtensions
{
    const double Tolerance = 1E-5;

    public static bool IsApproximatelyZero(this double x)
    {
        return Math.Abs(x) <= Tolerance;
    }

    public static bool IsApproximatelyOne(this double x)
    {
        return Math.Abs(x - 1) <= Tolerance;
    }
}
```

Usage in a method like `EaseInOut(double x)` would look like:

```
if (x.IsApproximatelyZero()) return 0;
if (x.IsApproximatelyOne()) return 1;
```

That way you avoid spending cycles doing things like `Math.Max(Math.Abs(x), Math.Abs(y))` when you know that `y` is zero.